src/modeler/AutoRest.Swagger/Validation/Core/RuleContext.cs

```diff
@@ -19,12 +19,16 @@ public class RuleContext
         /// Initializes a top level context for rules
         /// </summary>
         /// <param name="root"></param>
-        public RuleContext(ServiceDefinition root, Uri file) : this(null)
+        public RuleContext(ServiceDefinition root, Uri file, ServiceDefinitionMetadata metadata) : this(null)
         {
             this.Root = root;
             this.Value = root;
             this.File = file;
-            PopulateResourceTypes(root);
+            if (metadata.ServiceDefinitionDocumentType == ServiceDefinitionDocumentType.ARM)
```

Comment:
fair point, don't think anyone severely suffers because of this... main customer is still ARM, for data plane I also assume this goes faster since there is actually not much that looks like a resource. And even then, what's the actual overhead of this? Would be surprised if it took half a second on most Swaggers.